src/itemstackmetadata.cpp

```diff
@@ -29,15 +33,21 @@ void ItemStackMetadata::deSerialize(std::istream &is)
 	m_stringvars.clear();
 
 	if (!in.empty() && in[0] == DESERIALIZE_START) {
+		legacy_metadata = "";
+
 		Strfnd fnd(in);
 		fnd.to(1);
 		while (!fnd.at_end()) {
 			std::string name = fnd.next(DESERIALIZE_KV_DELIM_STR);
 			std::string var  = fnd.next(DESERIALIZE_PAIR_DELIM_STR);
-			m_stringvars[name] = var;
+			if (name == "") {
```

Comment:
as said, this will be fixed by #5176 
any other method will end up with one of the values overriding the other (data.metadata or data.fields[""])

> since there are mods that already saved (and still save) the metadata string into that field and expect getting the saved value back again.

Mods that do this shouldn't be considered, it's undocumented. They should use set_metadata instead.